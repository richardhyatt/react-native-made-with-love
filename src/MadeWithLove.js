import React from 'react';

import { Text, View, StyleSheet } from 'react-native';

const styles = StyleSheet.create({

  container: {

    flexDirection: 'row',
    backgroundColor: 'transparent',
    justifyContent: 'center',
  },
});

const MadeWithLove = ( { color, heartColor, place, style, textStyle, heartStyle } ) => {

  const _textStyle = {

    color,
    ...textStyle,
  };

  const _heartStyle = {

    color: heartColor,
    ...heartStyle,
  };

  return (
    <View style={[styles.container, style]}>
      <Text style={_textStyle}>Made with</Text>
      <Text style={_heartStyle}>{' ♥︎ '}</Text>
      <Text style={_textStyle}>{'in ' + place}</Text>
    </View>
  );
};

MadeWithLove.defaultProps = {

  color: 'black',
  heartColor: 'darkred',
  place: 'Toronto',
  style: {},
};

export default MadeWithLove;
