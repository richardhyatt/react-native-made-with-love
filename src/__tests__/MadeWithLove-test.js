import React from 'react';

import MadeWithLove from '../MadeWithLove.js';

import renderer from 'react-test-renderer';

test('normal operation without any changes', () => {

  const tree = renderer.create(<MadeWithLove/>).toJSON();
  expect(tree).toMatchSnapshot();
});

test('normal operation, place changed', () => {

  const tree = renderer.create(<MadeWithLove place={'New York'}/>).toJSON();
  expect(tree).toMatchSnapshot();
});
