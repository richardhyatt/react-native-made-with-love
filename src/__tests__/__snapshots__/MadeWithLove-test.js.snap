// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`normal operation without any changes 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "transparent",
        "flexDirection": "row",
        "justifyContent": "center",
      },
      Object {},
    ]
  }
>
  <Text
    style={
      Object {
        "color": "black",
      }
    }
  >
    Made with
  </Text>
  <Text
    style={
      Object {
        "color": "darkred",
      }
    }
  >
     ♥︎ 
  </Text>
  <Text
    style={
      Object {
        "color": "black",
      }
    }
  >
    in Toronto
  </Text>
</View>
`;

exports[`normal operation, place changed 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "transparent",
        "flexDirection": "row",
        "justifyContent": "center",
      },
      Object {},
    ]
  }
>
  <Text
    style={
      Object {
        "color": "black",
      }
    }
  >
    Made with
  </Text>
  <Text
    style={
      Object {
        "color": "darkred",
      }
    }
  >
     ♥︎ 
  </Text>
  <Text
    style={
      Object {
        "color": "black",
      }
    }
  >
    in New York
  </Text>
</View>
`;
